File: src/programmers/level1/체육복.java
package programmers.level1;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

public class 체육복 {
    public static void main(String[] args) {
        int n = 5;
        int[] lost = {2, 4};
        int[] reserve = {3, 1};
        solution(n, lost, reserve);
    }

    public static int solution(int n, int[] lost, int[] reserve) {
        List<Integer> lostStd = Arrays.stream(lost).boxed().collect(Collectors.toList());
        List<Integer> reserveStd = Arrays.stream(reserve).boxed().collect(Collectors.toList());

        for (int i : reserve) {
            if (lostStd.contains(i)) { // 여벌 체육복 도난
                lostStd.remove(Integer.valueOf(i));
                reserveStd.remove(Integer.valueOf(i));
            }
        }

        Collections.sort(reserveStd);

        for (int i : reserveStd) {
            if (lostStd.contains(i - 1))
                lostStd.remove(Integer.valueOf(i - 1));
            else if (lostStd.contains(i + 1))
                lostStd.remove(Integer.valueOf(i + 1));
        }

        return n - lostStd.size();
    }
}

File: src/programmers/javascript/level1/체육복.js
function solution(n, lost, reserve) {
    const onlyOne = new Array(n + 1).fill(false);
    
    for(let i = 1; i <= n; i++){
        if((reserve.includes(i) && lost.includes(i)) || (!lost.includes(i) && !reserve.includes(i))) { // 나 여벌이 있었는데 도난 당함. 못 빌려줌 또는 나 안빌려줘도 됨
            onlyOne[i] = true;
        }
    }
    
    for(let i = 1; i <= n; i++){
        if(onlyOne[i]) continue;
        
        if(reserve.includes(i)){ // 여벌 있음
            if(i - 1 !== 0 && lost.includes(i - 1) && !onlyOne[i - 1]){ // 앞 번호가 체육복이 없는 경우
                onlyOne[i - 1] = true;
            }else if(i + 1 !== n + 1 && lost.includes(i + 1) && !onlyOne[i + 1]){ // 뒷 번호가 체육복이 없는 경우
                onlyOne[i + 1] = true;
            }
            
            onlyOne[i] = true;
        }
    }
    
    return onlyOne.filter((it) => it).length;
}